### PR TITLE
fix: Upgrading code repo api client package to fix api response structure

### PR DESCRIPTION
**Pull request by vardanbansal-harness**
This fixes the issue where API response was getting wrapped in `content` key.

from

`return { content: data, ... } as TResponse;`

to

`return { data, ... } as TResponse;`

### CODE REVIEW

**Review by jessieyu777 — COMMENTED** [review]
Glad to see we can fetch data without content wrapper!

File: apps/gitness/src/pages/pipeline-list.tsx [hunk 1]
```diff
@@ -32,14 +32,13 @@ export default function PipelinesPage() {
...
-      // @ts-expect-error remove "@ts-expect-error" once type issue for "content" is resolved
-      placeholderData: {
-        content: [{ identifier: 'pipeline1' }, { identifier: 'pipeline2' }]
-      } as ListPipelinesOkResponse,
+      placeholderData: [{ identifier: 'pipeline1' }, { identifier: 'pipeline2' }],
       enabled: true
     }
   )
 
+  console.log(pipelines)
```
Comment (abhinavrastogi-harness) [inline apps/gitness/src/pages/pipeline-list.tsx:40]: unexpected console statement

File: apps/gitness/src/pages/repo/repo-list.tsx [hunk 2]
```diff
@@ -37,17 +38,16 @@ export default function ReposListPage() {
       data && (
         <RepoList
           LinkComponent={LinkComponent}
-          // @ts-expect-error remove "@ts-expect-error" once type issue for "content" is resolved
-          repos={data?.content?.map(repo => {
+          repos={data?.map((repo: RepoRepositoryOutput) => {
```
Comment (abhinavrastogi-harness) [inline apps/gitness/src/pages/repo/repo-list.tsx:41]: `repo: RepoRepositoryOutput`

this typecast should not be needed.. is the `data` not already an array of the correct type?

File: apps/gitness/src/pages/repo/repo-list.tsx [hunk 3]
```diff
@@ -37,17 +38,16 @@ export default function ReposListPage() {
...
             return {
               id: repo.id,
               name: repo.identifier,
               description: repo.description,
-              private: repo.private,
-              stars: repo.stars,
-              forks: repo.forks,
-              pulls: repo.pulls,
-              timestamp: repo.timestamp
+              private: repo.is_public,
```
Comment (abhinavrastogi-harness) [inline apps/gitness/src/pages/repo/repo-list.tsx:46]: this looks incorrect.. boolean should be inverse

File: apps/gitness/src/pages/repo/repo-list.tsx [hunk 4]
```diff
@@ -37,17 +38,16 @@ export default function ReposListPage() {
...
+              stars: 0,
```
Comment (abhinavrastogi-harness) [inline apps/gitness/src/pages/repo/repo-list.tsx:47]: hardcoded 0?

File: apps/gitness/src/pages/repo/repo-summary.tsx [hunk 5]
```diff
@@ -74,23 +68,17 @@ export const RepoSummary: React.FC = () => {
...
-    (): Map<string, OpenapiGetContentOutput['type']> =>
-      new Map(
-        // @ts-expect-error remove "@ts-expect-error" once type issue for "content" is resolved
-        repoDetailsContent?.content?.entries.map(entry => [entry.path, entry.type])
-      ),
-    // @ts-expect-error remove "@ts-expect-error" once type issue for "content" is resolved
-    [repoDetailsContent?.content?.entries]
-  )
+  const repoEntryPathToFileTypeMap = useMemo((): Map<string, OpenapiGetContentOutput['type']> => {
+    if (repoDetails?.content?.entries?.length === 0) return new Map()
```
Comment (abhinavrastogi-harness) [inline apps/gitness/src/pages/repo/repo-summary.tsx:72]: this response actually contains the keyword `content`?

File: apps/gitness/src/pages/repo/repo-list.tsx (hunk 4 above)
Comment (vardanbansal-harness) [reply apps/gitness/src/pages/repo/repo-list.tsx:47]: Removed it, also made `stars` attribute optional for now. Currently, repo listing api does not have `star` count in it. I  have also added it to the API requirement doc.

File: apps/gitness/src/pages/repo/repo-list.tsx (hunk 3 above)
Comment (vardanbansal-harness) [reply apps/gitness/src/pages/repo/repo-list.tsx:46]: Fixed it.

File: apps/gitness/src/pages/pipeline-list.tsx (hunk 1 above)
Comment (vardanbansal-harness) [reply apps/gitness/src/pages/pipeline-list.tsx:40]: Removed it.

File: apps/gitness/src/pages/repo/repo-summary.tsx (hunk 5 above)
Comment (vardanbansal-harness) [reply apps/gitness/src/pages/repo/repo-summary.tsx:72]: Yes, it's defined as below

`export interface OpenapiGetContentOutput {
    content?: OpenapiContent;
    latest_commit?: TypesCommit;
    name?: string;
    path?: string;
    sha?: string;
    type?: OpenapiContentType;
}`

File: apps/gitness/src/pages/repo/repo-list.tsx (hunk 2 above)
Comment (vardanbansal-harness) [reply apps/gitness/src/pages/repo/repo-list.tsx:41]: This is actually a weird scenario. The API returns a response of type `TypesRepository[]` by definition. But the actual response returned by the api endpoint is of type `RepoRepositoryOutput`. I observed it when I was looking for `is_public` attribute, which isn't present on type `TypesRepository`.

Comment (vardanbansal-harness) [reply apps/gitness/src/pages/repo/repo-list.tsx:41]: I will start a thread for this.

**Review by cjlee01 — APPROVED** [review]

**Pull request merged**